dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +63,120 @@ def __init__(
     def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]: ...
 
 
+class ModelItemsNormalizer(ItemsNormalizer):
+    def _add_or_replace_column(
+        self,
+        parsed_select: sqlglot.exp.Select,
+        column_name: Literal["_dlt_load_id", "_dlt_id"],
+        alias_expression: sqlglot.exp.Alias,
+        root_table_name: str,
+    ) -> Optional[TSchemaUpdate]:
+        """
+        Adds or replaces a column in the SELECT clause of a parsed SQL query and
+        returns a single TSchemaUpdate if a new column was added, otherwise None.
+        """
+        schema = self.schema
+        schema_update: TSchemaUpdate = {}
+
+        column_replaced = False
+        for i, select in enumerate(parsed_select.selects):
+            # Check if the column already exists
+            select_alias = (
+                select.alias_or_name.lower()
+                if isinstance(select, sqlglot.exp.Alias)
+                else (
+                    select.output_name.lower() if isinstance(select, sqlglot.exp.Column) else None
+                )
+            )
+
+            if select_alias == column_name:
+                # Replace the existing column
+                parsed_select.selects[i] = alias_expression
+                column_replaced = True
+                break
+
+        if not column_replaced:
+            # Append the column if it doesn't exist
+            parsed_select.selects.append(alias_expression)
+
+            partial_table = normalize_table_identifiers(
+                {
+                    "name": root_table_name,
+                    "columns": {
+                        column_name: (
+                            dlt_id_column() if column_name == "_dlt_id" else dlt_load_id_column()
+                        )
+                    },
+                },
+                schema.naming,
+            )
+
+            table_updates = schema_update.setdefault(root_table_name, [])
+            table_updates.append(partial_table)
+
+            return schema_update
+
+        return None
+
+    def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]:
+        with self.normalize_storage.extracted_packages.storage.open_file(
+            extracted_items_file, "r"
+        ) as f:
+            select_dialect, select_statement = read_dialect_and_sql(
+                file_obj=f,
+                fallback_dialect=self.config.destination_capabilities.sqlglot_dialect,  # caps are available at this point
+            )
+
+        parsed_select = sqlglot.parse_one(select_statement, read=select_dialect)
+        parsed_select = cast(sqlglot.exp.Select, parsed_select)
+
+        schema_updates = []
+
+        if self.config.json_normalizer.add_dlt_load_id:
```

Comment:
I am not quite sure if part of this task is also to harmonize the normalizer behavior across loader_file_formats, if it is, we can ditch the `json_normalizer` and `parquet_normalizer` on `NormalizeConfiguration` and just use on normalizer config for all three normalizers. If we do not harmonize, then I'd instead add a `model_normalizer`config that has the same defaults as the `json_normalizer`.